src/VisualStudio/Core/Def/Implementation/Workspace/VisualStudioDocumentNavigationService.cs

```diff
@@ -43,19 +43,22 @@ internal sealed class VisualStudioDocumentNavigationService : ForegroundThreadAf
         private readonly IVsEditorAdaptersFactoryService _editorAdaptersFactoryService;
         private readonly IVsRunningDocumentTable4 _runningDocumentTable;
         private readonly IThreadingContext _threadingContext;
+        private readonly Lazy<SourceGeneratedFileManager> _sourceGeneratedFileManager;
 
         [ImportingConstructor]
         [Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
         public VisualStudioDocumentNavigationService(
             IThreadingContext threadingContext,
             SVsServiceProvider serviceProvider,
-            IVsEditorAdaptersFactoryService editorAdaptersFactoryService)
+            IVsEditorAdaptersFactoryService editorAdaptersFactoryService,
+            Lazy<SourceGeneratedFileManager> sourceGeneratedFileManager /* lazy to avoid circularities */)
```

Comment:
As of this commit, the SourceGeneratedFileManager imports VisualStudioDocumentNavigationService, and the latter also imports the former. This isn't great but longer term I'm hoping the "interesting" bits of SourceGeneratedFileManager move down a few layers so they can be shared with other hosts, and the leftovers merge into the VisualStudioDocumentNavigationService.